internal/gengapic/genrest.go

```diff
@@ -838,12 +838,19 @@ func (g *generator) unaryRESTCall(servName string, m *descriptor.MethodDescripto
 	p("  return nil, err")
 	p("}")
 	p("")
-	p("unm := protojson.UnmarshalOptions{AllowPartial: true, DiscardUnknown: true}")
-	p("rsp := &%s.%s{}", outSpec.Name, outType.GetName())
-	p("")
-	p("if err := unm.Unmarshal(buf, rsp); err != nil {")
-	p("  return nil, maybeUnknownEnum(err)")
-	p("}")
+	if outSpec.Name == "httpbodypb" && outType.GetName() == "HttpBody" {
+		p("rsp := &httpbodypb.HttpBody{")
+		p("  Data: buf,")
+		p("  ContentType: httpRsp.Header[\"Content-Type\"][0],")
```

Comment:
This is brittle - clients will crash if there is no Content-Type header.

Reply:
Yeah, let's make this more robust, see my suggestion.

Reply:
Done, thanks!